test/e2e/network/network_policy.go

```diff
@@ -821,49 +821,39 @@ var _ = SIGDescribe("NetworkPolicy [LinuxOnly]", func() {
+		ginkgo.It("should deny ingress access to updated pod [Feature:NetworkPolicy]", func() {
 			const allowedPort = 80
-			ginkgo.By("Creating a network policy for the server which allows traffic from client-a-updated.")
+			ginkgo.By("Creating a network policy for the server which denies all traffic.")
 			policy := &networkingv1.NetworkPolicy{
 				ObjectMeta: metav1.ObjectMeta{
-					Name: "allow-pod-b-via-pod-selector",
+					Name: "deny-ingress-via-isolated-label-selector",
 				},
 				Spec: networkingv1.NetworkPolicySpec{
 					PodSelector: metav1.LabelSelector{
 						MatchLabels: map[string]string{
 							"pod-name": podServerLabelSelector,
 						},
-					},
-					Ingress: []networkingv1.NetworkPolicyIngressRule{{
-						From: []networkingv1.NetworkPolicyPeer{{
-							PodSelector: &metav1.LabelSelector{
-								MatchExpressions: []metav1.LabelSelectorRequirement{{
-									Key:      "pod-name",
-									Operator: metav1.LabelSelectorOpDoesNotExist,
-								}},
-							},
+						MatchExpressions: []metav1.LabelSelectorRequirement{{
+							Key:      "isolated",
+							Operator: metav1.LabelSelectorOpExists,
 						}},
-					}},
+					},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+					Ingress:     []networkingv1.NetworkPolicyIngressRule{},
 				},
 			}
 
 			policy, err := f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policy)
 			framework.ExpectNoError(err, "Error creating Network Policy %v: %v", policy.ObjectMeta.Name, err)
 			defer cleanupNetworkPolicy(f, policy)
 
-			ginkgo.By(fmt.Sprintf("Creating client pod %s that should not be able to connect to %s.", "client-a", service.Name))
-			podClient := createNetworkClientPod(f, f.Namespace, "client-a", service, allowedPort)
-			defer func() {
-				ginkgo.By(fmt.Sprintf("Cleaning up the pod %s", podClient.Name))
-				if err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Delete(podClient.Name, nil); err != nil {
-					framework.Failf("unable to cleanup pod %v: %v", podClient.Name, err)
-				}
-			}()
-			checkNoConnectivity(f, f.Namespace, podClient, service)
-
-			ginkgo.By(fmt.Sprintf("Updating client pod %s that should successfully connect to %s.", podClient.Name, service.Name))
-			podClient = updateNetworkClientPodLabel(f, f.Namespace, podClient.Name, "replace", "/metadata/labels", map[string]string{})
-			checkConnectivity(f, f.Namespace, podClient, service)
+			// Client can connect to service when the server pod is not selected by the network policy.
+			testCanConnect(f, f.Namespace, "client-a", service, allowedPort)
+
+			// Client cannot connect to service after updating the server pod's labels to match the network policy.
+			ginkgo.By(fmt.Sprintf("Updating server pod %s to be selected by network policy %s.", podServer.Name, policy.Name))
+			updateNetworkClientPodLabel(f, f.Namespace, podServer.Name, "add", "/metadata/labels", map[string]string{"isolated": ""})
```

Comment:
Use updateNetworkClientPodLabel to change a server pod label is confusing. Could you rename the function to a generic one?

Reply:
I didn't notice the function name, thanks for catching it, done.